packages/vscode/src/extension.ts

```diff
@@ -57,6 +60,32 @@ async function restartClients(): Promise<void> {
   await Promise.all([...clients.values()].map((client) => client.restart()));
 }
 
+async function sortImports(editor: TextEditor): Promise<void> {
+  let workspaceFolder = workspace.getWorkspaceFolder(editor.document.uri);
+  if (!workspaceFolder) {
+    return;
+  }
+
+  let client = clients.get(workspaceFolder.uri.fsPath);
+  let request = requestKey<typeof SortImportsRequest>('glint/sortImports');
+  const edits = await client?.sendRequest(request, { uri: editor.document.uri.toString() });
+
+  if (!edits) {
+    return;
+  }
+
+  const workspaceEdit = new WorkspaceEdit();
+
+  for (const edit of edits) {
+    const start = new Position(edit.range.start.line, edit.range.start.character);
+    const end = new Position(edit.range.end.line, edit.range.end.character);
+    const range = new Range(start, end);
+    workspaceEdit.replace(editor.document.uri, range, edit.newText);
```

Comment:
Is there a reason you need to reconstruct a fresh `Range` object here?

```suggestion
    workspaceEdit.replace(editor.document.uri, edit.range, edit.newText);
```

Reply:
The types for `vscode` (eg. [Range](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/vscode/index.d.ts#L399)) differ from `vscode-language-server` (eg. [Range](https://github.com/microsoft/vscode-languageserver-node/blob/main/types/src/main.ts#L181)) slightly, so we have modify the response to match what the client expects. Kind of annoying :shrug: 

We could return a value that matches a `vscode` type, but then the language server wouldn't be editor-agnostic.

vscode's built-in extension adds [helper methods to `Range`](https://github.com/microsoft/vscode/blob/main/extensions/typescript-language-features/src/typeConverters.ts#L75) to perform the conversion before [calling `applyEdit`](https://github.com/microsoft/vscode/blob/243e6cec5096b301259ca1f780e50b135f308d33/extensions/typescript-language-features/src/languageFeatures/organizeImports.ts#L106). We could do the same, but it seemed a bit premature?

Reply:
Ah! Lovely. Thanks, `vscode` 😏 

It looks like `Range` has a secondary construct signature that accepts four numbers instead of requiring the intermediary `Position` objects. Maybe that's a slight improvement?

```ts
    const { start, end } = edit.range;
    const range = new Range(start.line, start.character, end.line, end.character);
    workspaceEdit.replace(editor.document.uri, range, edit.newText);
```